### PR TITLE
#17674 use database datatype when case not equivalent

### LINKED ISSUE #17674: Invalid type int8 instead of Int8 when export from PostgreSQL and import to ClickHouse
### Description

Can not transfer data from PostgreSQL to ClickHouse, because generated DDL for create new table has invalid type 'int8' instead of 'Int8'

### DBeaver Version

CE Version 22.2.0.202209051344

### Operating System

macOS Monterey 12.6

### Database and driver

Connection Drivers:
PostgreSQL standard driver
Yandex ClickHouse legacy driver (server version < 21.3)

### Steps to reproduce

1. In Postgres create table t1 with a column typed int8
2. Insert data to table t1 (insert into t1 values (1)) 
3. Export data by context menu from PostgreSQL to ClickHouse

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Destrolaric**
java_3hpkuVV9aS

Closes #17674

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java [hunk 1]
```diff
@@ -425,6 +425,8 @@ public static String mapTargetDataType(DBSObject objectContainer, DBSTypedObject
             }
             if (dataType != null) {
                 dataKind = dataType.getDataKind();
+                // Datatype caches ignore case, but we probably should use it with the original case
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java:432]: But usually original <> target, but source. And in this case target type name is used.

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java:432]: (About the comment)

**Comment by serge-rider** [comment]
Strange fix. A couple of lines before the change we set typeName to targetType name. In this fix we use source type name. 

It will break all data transfers in all databases because idea is to use target database type.

**Comment by uslss** [comment]
verified

**Pull request merged**